internal/controller/nimcache_controller.go

```diff
@@ -463,7 +464,7 @@ func (r *NIMCacheReconciler) reconcileJobStatus(ctx context.Context, nimCache *a
 			return fmt.Errorf("failed to get selected profiles: %w", err)
 		}
 
-		if len(selectedProfiles) > 0 {
+		if len(selectedProfiles) > 0 && !utils.ContainsElement(selectedProfiles, "all") {
```

Comment:
Does `selectedProfiles` represent a concrete type. Does it make sense to add a function like `IsAll()` to capture this logic?

Reply:
yes, it is a concrete type, but adding a utility call for this makes sense.

Reply:
double checked again and this is confined to NIMCache type as "all" profiles is not used during serving. Keeping the code as-is.